test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1

```diff
@@ -109,49 +110,103 @@ baz
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 0
+            tail = 3})
+        expectedLength = 3
+        expectedContent = "Hello2,World2","Hello3,World3","Hello4,World4"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 1
+            tail = 3})
+        expectedLength = 3
+        expectedContent = "Hello2,World2","Hello3,World3","Hello4,World4"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 99999
+            tail = 3})
+        expectedLength = 3
+        expectedContent = "Hello2,World2","Hello3,World3","Hello4,World4"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 99999
+            tail = 3})
+        expectedLength = 3
+        expectedContent = "Hello2,World2","Hello3,World3","Hello4,World4"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 2
+            tail = 3})
+        expectedLength = 2
+        expectedContent = ("Hello2,World2","Hello3,World3"), "Hello4,World4"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            readcount = 2
+            tail = 2})
+        expectedLength = 2
+        expectedContent = "Hello3,World3","Hello4,World4"
+        }
+    ) {#[DRT]
+        param($GetContentParams, $expectedLength, $expectedContent)
+        Set-Content -Path $testPath $testContent
+        $result = Get-Content -Path $GetContentParams.path -Readcount $GetContentParams.readcount -Tail $GetContentParams.tail
+        $result.Length | Should Be $expectedLength
+        $result -eq $expectedContent
+    }
+    It "should Get-Content with a variety of -Delimiter and -Tail values" -TestCases @(
+        @{
+        GetContentParams = @(@{path = $testPath
+            delimiter = ","
+            tail = 2})
+        expectedLength = 2
+        expectedContent = "World3${nl}Hello4", "World4${nl}"
+        }
+        @{
+        GetContentParams = @(@{path = $testPath
+            delimiter = "o"
+            tail = 3})
+        expectedLength = 3
+        expectedContent = "rld3${nl}Hell", '4,W', "rld4${nl}"
+        }
+    ) {#[DRT]
+        param($GetContentParams, $expectedLength, $expectedContent)
+        Set-Content -Path $testPath $testContent
+        $result = Get-Content -Path $GetContentParams.path -Delimiter $GetContentParams.delimiter -Tail $GetContentParams.tail
+        $result.Length | Should Be $expectedLength
+        $result | Should BeExactly $expectedContent
+    }
+    It "should Get-Content with a variety of -Tail values and -AsByteStream parameter" -TestCases @(
+        @{
+        GetContentParams = @(@{path = $testPath
+            tail = 10})
+        expectedLength = 10
+        expectedWindowsContent =         52, 44, 87, 111, 114, 108, 100, 52, 13, 10
+        expectedNotWindowsContent = 111, 52, 44, 87, 111, 114, 108, 100, 52, 10
```

Comment:
Don't understand why on non-Windows there is a leading 111 character?  Seems like the only difference is the newline differences.